src/openrct2/peep/Guest.cpp

```diff
@@ -7356,3 +7356,115 @@ bool Guest::HasItem(ShopItem peepItem) const
 {
     return GetItemFlags() & EnumToFlag(peepItem);
 }
+
+static bool IsThoughtShopItemRelated(const PeepThoughtType type)
+{
+    switch (type)
+    {
+        case PeepThoughtType::AlreadyGot:
+        case PeepThoughtType::HaventFinished:
+        case PeepThoughtType::CantAffordItem:
+            return true;
+        default:
+            break;
+    }
+    return false;
+}
+
+void Guest::RemoveRideFromMemory(ride_id_t rideId)
+{
+    uint8_t ride_id_bit = rideId % 8;
+    uint8_t ride_id_offset = rideId / 8;
+
+    // clear ride from potentially being in RidesBeenOn
+    RidesBeenOn[ride_id_offset] &= ~(1 << ride_id_bit);
```

Comment:
Access to RidesBeenOn should be in a method of its own, I’d say, as accessing bits in it is not trivial. It also makes it easier to refactor (there is a refactor from Duncan coming up).

Reply:
Yeah I spoke to Duncan, this will be replaced with his methods later.